fwoa-tools/src/export-script.py

```diff
@@ -0,0 +1,293 @@
+filtered_dates_dyn_frame = Filter.apply(frame = filtered_group_reference_frame,
+                           f = lambda x:
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") > datetime_since and
+                           datetime.strptime(x["meta"]["lastUpdated"], "%Y-%m-%dT%H:%M:%S.%fZ") <= datetime_transaction_time
+                          )
+
+print('Start filtering by documentStatus and resourceType')
+# Filter by resource listed in Type and with correct STATUS
+type_list = None if type == None else set(type.split(','))
+valid_document_state_to_be_read_from = {'AVAILABLE','LOCKED', 'PENDING_DELETE'}
+if (exportDeletedResources):
+    valid_document_state_to_be_read_from = {'AVAILABLE','LOCKED', 'PENDING_DELETE', 'DELETED'}
+    
+filtered_dates_resource_dyn_frame = Filter.apply(frame = filtered_dates_dyn_frame,
+                                    f = lambda x:
+                                    x["documentStatus"] in valid_document_state_to_be_read_from if type_list is None
+                                    else x["documentStatus"] in valid_document_state_to_be_read_from and x["resourceType"] in type_list
+                          )
+
+
+# Drop fields that are not needed
+print('Dropping fields that are not needed')
+data_source_cleaned_dyn_frame = DropFields.apply(frame = filtered_dates_resource_dyn_frame, paths = ['documentStatus', 'lockEndTs', 'vid', '_references', '_tenantId', '_id', '_subscriptionStatus'])
+
+def add_dup_resource_type(record):
+    # if we want deleted resources, we want to export by resourceType+id
+    if (exportDeletedResources):
+        record["partitionKeyDup"] = record["resourceType"] + record["id"]
+    else:
+        record["partitionKeyDup"] = record["resourceType"]
+    return record
+
+# Create duplicated column so we can use it in partitionKey later
+data_source_cleaned_dyn_frame = data_source_cleaned_dyn_frame.map(add_dup_resource_type)
+
+# To export one S3 file per resourceType, we repartition(1)
+data_source_cleaned_dyn_frame = data_source_cleaned_dyn_frame.repartition(1)
+
+if len(data_source_cleaned_dyn_frame.toDF().head(1)) == 0:
+    print('No resources within requested parameters to export')
+else:
+    print('Writing data to S3')
+    # Export data to S3 split by resourceType
+    glueContext.write_dynamic_frame.from_options(
+        frame = data_source_cleaned_dyn_frame,
+        connection_type = "s3",
+        connection_options = {
+            "path": "s3://" + bucket_name + "/" + job_id,
+            "partitionKeys": ["partitionKeyDup"],
+        },
+        format = "json"
+    )
+
+    # Rename exported files into ndjson files
+    print('Renaming files')
+    client = boto3.client('s3')
+
+    response = client.list_objects(
+        Bucket=bucket_name,
+        Prefix=job_id,
+    )
+
+    regex_pattern = '\/partitionKeyDup=(\w+)(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})\/run-\d{13}-part-r-(\d{5})'
```

Comment:
Can you add beginning and end anchors here for exact pattern matching https://quip-amazon.com/9v4sA7lCuHSb/Regular-Expressions#temp:C:ZNYb552e89d3f9a484195d5c7ddc

Reply:
Updated to add the end anchor. Unfortunately as this is part of an s3 uri, the pattern is only looking for the end of item key. The rest of the string would contain the s3 bucket and folder name as well, so we can't add the start anchor here. Verified that the export completes successfully with the end anchor though!